src/CliqStateMachine.jl

```diff
@@ -532,29 +529,28 @@ function attemptCliqInitDown_StateMachine(csmc::CliqStateMachineContainer)
   # determine if more info is needed for partial
   sdims = getCliqVariableMoreInitDims(csmc.cliqSubFg, csmc.cliq)
   updateCliqSolvableDims!(csmc.cliq, sdims, csmc.logger)
-  infocsm(csmc, "8a, attemptCliqInitD., updated clique solvable dims")
+  infocsm(csmc, "8a, attemptCliqInitD., updated clique solvable dims and unlock")
 
-  # infocsm(csmc, "8a, attemptCliqInitD., deleted msg factors and unlockUpStatus!")
   # unlock
   unlockUpStatus!(prnt) # TODO XY ????
   infocsm(csmc, "8a, attemptCliqInitD., unlocked")
 
 
   # FIXME try split CSM here, need replacement for dwinmsgs
   dwnkeys_ = ls(csmc.cliqSubFg, tags=[:DOWNWARD_COMMON;]) # FIXME THIS DOESNT WORK YET!
-  # @assert intersect(dwnkeys, dwnkeys_) == length(dwnkeys) "split dwnkeys_ is not the same, $dwnkeys, and $dwnkeys_"
-
+  # FIXME, why is this breaking the IIF and RoME tests?
+  @assert intersect(dwnkeys, dwnkeys_) == length(dwnkeys) "split dwnkeys_ is not the same, $dwnkeys, and $dwnkeys_"
```

Comment:
I tried a bit, but don't follow completely. 

- the assert `intersect(dwnkeys, dwnkeys_) == length(dwnkeys)` is a set and a length
- `DOWNWARD_COMMON` is added to factors, so I thought the idea was to get the variables attached to the factors with that tag.

Here is what I tried, it works for a few, but there are cases with no DOWNWARD_COMMON tag at all.

```julia
  # FIXME try split CSM here, need replacement for dwinmsgs
  @info "DOWNWADS_COMMON $(lsf(csmc.cliqSubFg, tags=[:DOWNWARD_COMMON;])) --- $dwnkeys"
  dwnkeys_ = Symbol[]
  union!(dwnkeys_, getNeighbors.(csmc.cliqSubFg, lsf(csmc.cliqSubFg, tags=[:DOWNWARD_COMMON;]))...)
   # FIXME THIS DOESNT WORK YET!
  # FIXME, why is this breaking the IIF and RoME tests?
  @assert issetequal(dwnkeys, dwnkeys_) "split dwnkeys_ is not the same, $dwnkeys, and $dwnkeys_"
```

Reply:
Ahh, I was close. will leave down messages to you.